assets/src/components/v2/persistent_wrapper.tsx

```diff
@@ -27,16 +26,14 @@ const PersistentWrapper: ComponentType<Props> = ({
   useEffect(() => {
     if (isFinished) {
       setVisibleData(data);
-      setRenderKey((n) => n + 1);
       setIsFinished(false);
     }
-  }, [lastFetch]);
+  }, [isFinished]);
```

Comment:
Triggering the data refresh on `isFinished` instead of `lastFetch` allows this Wrapper to work the same way for both interval and refresh based pagination.

Reply:
This is a great callout! That's even simpler and I think should work